src/webgpu/api/validation/createBindGroupLayout.spec.ts

```diff
@@ -68,101 +70,89 @@ g.test('visibility_and_dynamic_offsets')
     }, !success);
   });
 
-g.test('min_buffer_binding_size')
+g.test('non_required_parameter_combined_with_type_separately')
   .params(
     params()
       .combine(poptions('type', kBindingTypes))
-      .combine(poptions('minBufferBindingSize', [undefined, 0, 4]))
+      .combine([
+        ...poptions('minBufferBindingSize', [undefined, 0, 4]),
+        ...poptions('textureComponentType', [undefined, ...kTextureComponentTypes]),
+        ...poptions('multisampled', [undefined, false, true]),
+        ...poptions('viewDimension', [undefined, ...kTextureViewDimensions]),
+        ...poptions('storageTextureFormat', [undefined, ...kTextureFormats]),
+      ])
```

Comment:
Um. It is not necessary to add "undefined" to the parameter if it is not tested. However, we can't test "undefined" value for each optional parameter when we are explicitly testing that parameter right now if we don't explicitly add "undefined".

Reply:
I don't understand your last comment. My point (sorry it was unclear) is that all of the following test cases are the same:
- `{minBufferBindingSize: undefined}`
- `{textureComponentType: undefined}`
- ...
- `{}`

so we should only have one of them (which is `{}` at the top of the suggestion).

Reply:
Oops. I missed {} at the top you mentioned. Now I tried to add it and removed "undefined" for each parameter, but it failed to build via "npx grunt pre". 

The current one should be good. For each binding type, it will test each optional parameters with values listed in bracket. For example, the test cases for one binding type is: 
minBufferBindingSize=undefined
minBufferBindingSize=0
minBufferBindingSize=4
textureComponentType=undefined
textureComponentType="float"
textureComponentType="sint"
textureComponentType="uint"
...

I think the parameters that are not showed should be "undefined", so the test cases are exactly what we want. WDYT? @kainino0x

Reply:
ugh, looks like this found another corner case of the type metaprogramming that is broken. I'll see if there's an obvious issue, otherwise let's forget about this suggestion.

Reply:
The issue actually crops up before TypeScript reaches our type metaprogramming:
```typescript
const x = [{ a: 0 }];
const y = [{}, ...x];
type Y = typeof y;
```
gives Y as `{}[]`.

Found a complex workaround and a simple workaround. Uploaded the simple one here and the complex one in #248.

Reply:
Looks like the current simple workaround is incorrect. I just verified, and the test cases for each binding types are:
type="uniform-buffer"  >>> This is not necessary? 
minBufferBindingSize=0
minBufferBindingSize=4
textureComponentType="float"
textureComponentType="sint"
textureComponentType="uint"
...

We don't have explicit tests of "undefined" for each optional parameter. 

My previous tests are: 
minBufferBindingSize=undefined  >>> no test like this
minBufferBindingSize=0
minBufferBindingSize=4
textureComponentType=undefined   >>> no test like this
textureComponentType="float"
textureComponentType="sint"
textureComponentType="uint"
...

WDYT? @kainino0x. I can revise back in this patch https://github.com/gpuweb/cts/pull/249 if needed.

Reply:
The current workaround might be OK too. I just realized that this test
type="uniform-buffer"
means all optional parameters are undefined. The only difference is that my previous test will test against explicit "undefined" for each parameter. Maybe one test of implicit "undefined" for all optional parameters is enough, just like what it is right now.

Reply:
yeah, because of the way we feed the parameters into the test function the cases {minBufferBindingSize: undefined}, {textureComponentType: undefined}, etc. are all identical. This code doesn't test the difference between `undefined` and not specifying a member at all. This is not very important, because WebIDL declares that they are identical, so as long as the browser's WebIDL implementation is correct, they'll be the same.

Reply:
OK. Thanks for your explanation, Kai.